### PR TITLE
Rehaul: Google OAuth

### PR DESCRIPTION
**Pull request by Pranay105**
- Added sub-headings for clarity
- grammar fix

### CODE REVIEW

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 1]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+  ![Click to expand](</img/Google_OAuth_Consent_1.png>)
 
-**2. Configure the consent screen with the domain on which you want to host Appsmith**
+2. Configure the consent screen with the domain you want to host Appsmith.
 
-![Click to expand](</img/Google_OAuth_Consent.png>)
+  ![Click to expand](</img/Google_OAuth_Consent.png>)
 
-**3. Navigate to the** [**credentials section**](https://console.cloud.google.com/apis/credentials) **and create new OAuth Credentials**
+3. Navigate to the [**credentials**](https://console.cloud.google.com/apis/credentials) and create new OAuth Credentials.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:24]:
```suggestion
3. Navigate to the Credentials screen and click **Create Credentials**, select **OAuth client ID** from the options to create OAuth credentials.
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 2]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
 # Google OAuth
 
-Google OAuth is one of the most widely used custom authentication integrations. Appsmith allows you to integrate with Google for authenticating users to your apps.
+Appsmith allows you to integrate with Google to authenticate users to your apps.
 
-![Click to expand](</img/Google_Login.png>)
+## Configure Google Cloud
 
-To enable Google Sign in, login to your [google cloud console](https://console.cloud.google.com/)
+To enable Google Sign in, log in to your [google cloud console](https://console.cloud.google.com/) and follow the steps below:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:15]: I think we should also tell them to create a new project and such.

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 3]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+  ![Click to expand](</img/Google_OAuth_Consent.png>)
 
-**3. Navigate to the** [**credentials section**](https://console.cloud.google.com/apis/credentials) **and create new OAuth Credentials**
+3. Navigate to the [**credentials**](https://console.cloud.google.com/apis/credentials) and create new OAuth Credentials.
 
-![Click to expand](</img/Google_OAuth_Creds.png>)
+  ![Click to expand](</img/Google_OAuth_Creds.png>)
 
-**4. Configure the OAuth Credentials for a web application**
+4. Select the application type as a Web application. Update the following fields:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:28]:
```suggestion
4. Select Web application under the Application Type dropdown list. Update the following fields:
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 4]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
 
-**Javascript Origins:** the domain on which you are hosting Appsmith with HTTPS **`(ex: https://app.appsmith.com)`**
+    1. **Javascript Origins:** the domain on which you are hosting Appsmith with HTTPS **`(ex: https://app.appsmith.com)`**
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:30]: It should be Authorized JavaScript origins: 

also don't think this is HTTPS. please clarify this 
Screenshot 2023-03-09 at 19 44 35

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 5]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+    2. **Redirect URIs:** append **/login/oauth2/code/google** to your Javascript origins **`(ex: https://app.appsmith.com/login/oauth2/code/google)`**
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:31]:
```suggestion
    2. **Authorized redirect URIs:** Append the path `/login/oauth2/code/google` that users should be redirected to after they have authenticated with Google (Eg: `https://app.appsmith.com/login/oauth2/code/google`)
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 6]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
 
-**Redirect URIs:** append **/login/oauth2/code/google** to your Javascript origins **`(ex: https://app.appsmith.com/login/oauth2/code/google)`**
+5. Click Create. You can get the Client ID and Client secret under "OAuth 2.0 Client IDs" in the Credentials tab.
 
-![Click to expand](</img/Google_Oauth_Creds_2.png>)
+  ![Click to expand](</img/Google_Oauth_Creds_2.png>)
 
-**5. Update the**[ **instance configuration**](../) **with the following fields**:
+## Configure your instance
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:37]:
```suggestion
## Configure Appsmith instance  
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 7]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+5. Click Create. You can get the Client ID and Client secret under "OAuth 2.0 Client IDs" in the Credentials tab.
 
-![Click to expand](</img/Google_Oauth_Creds_2.png>)
+  ![Click to expand](</img/Google_Oauth_Creds_2.png>)
 
-**5. Update the**[ **instance configuration**](../) **with the following fields**:
+## Configure your instance  
 
-  * **Configure via UI (recommended)**:
+Update the [**instance configuration**](../) with the following fields:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:39]: There are two ways to configure the self-hosted Appsmith instance: Admin Settings(link to section below)
and Environment Variables(link to section below)

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 8]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
 
-  On your self-hosted instance, navigate to the [Admin Settings page](/getting-started/setup/instance-configuration/). Under the **Authentication** tab, you can find fields for providing your Client ID and Client Secret generated in the previous steps.
+1. **Configure via UI (recommended):**  On your self-hosted instance, navigate to the [Admin Settings page](/getting-started/setup/instance-configuration/). Under the **Authentication** tab, you can find fields for providing your Client ID and Client Secret generated in the previous steps.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:41]:
```suggestion
1. **On Admin Settings screen (recommended):**  On your self-hosted instance, navigate to the [Admin Settings page](/getting-started/setup/instance-configuration/). Under the **Authentication** tab,  provide the Client ID and Client Secret generated in the previous steps.
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 9]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
 
-  ![](/img/as_google_auth_config.png)
+![](/img/as_google_auth_config.png)
 
-  Once you've entered your provided the Client ID and Client Secret, click the **Save & Restart** button at the bottom of the page. This prompts your Appsmith instance with the new configuration in effect.
+Once you've entered your Client ID and Client Secret, click the **Save & Restart** button at the bottom of the page. This prompts your Appsmith instance with the new configuration in effect.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:45]: move this above the image so users don't miss this

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 10]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
 
-  * **Configure via docker.env file**:
-    
-    On the machine hosting the Appsmith instance, find your `docker.env` file located in `<project-root>/stacks/configuration/`. Update the values for the following keys:
+2. **Configure via docker.env file**: On the machine hosting the Appsmith instance, find your `docker.env` file located in `<project-root>/stacks/configuration/`. Update the values for the following keys:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:47]:
```suggestion
2. **Using environment variables**: Update the values for the following keys in your configuration file.
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 11]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+  APPSMITH_OAUTH2_GOOGLE_CLIENT_SECRET=YOUR_CLIENT_SECRET
+  APPSMITH_SIGNUP_ALLOWED_DOMAINS=exampledomain.com
+  # ******************************
+   ```
 
     * Configure the `APPSMITH_OAUTH2_GOOGLE_CLIENT_ID` and `APPSMITH_OAUTH2_GOOGLE_CLIENT_SECRET` fields with the client ID and client secret generated in the step above
     * Configure `APPSMITH_SIGNUP_ALLOWED_DOMAINS` with a second-level domain name **`(ex: abc.com)`**, to only allow users with an email address from that domain name to log in **`(ex: john@abc.com)`**
-      * **Note**: you can provide several domain names using a comma-separated list
+    * **Note**: you can provide several domain names using a comma-separated list
     * [Restart the Appsmith instance](../)
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:61]:
```suggestion
    * Restart the Appsmith instance.
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 12]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+  # Example configuration in docker.env file
+  # ******** Google OAuth ********
+  APPSMITH_OAUTH2_GOOGLE_CLIENT_ID=YOUR_CLIENT_ID
+  APPSMITH_OAUTH2_GOOGLE_CLIENT_SECRET=YOUR_CLIENT_SECRET
+  APPSMITH_SIGNUP_ALLOWED_DOMAINS=exampledomain.com
+  # ******************************
+   ```
 
     * Configure the `APPSMITH_OAUTH2_GOOGLE_CLIENT_ID` and `APPSMITH_OAUTH2_GOOGLE_CLIENT_SECRET` fields with the client ID and client secret generated in the step above
     * Configure `APPSMITH_SIGNUP_ALLOWED_DOMAINS` with a second-level domain name **`(ex: abc.com)`**, to only allow users with an email address from that domain name to log in **`(ex: john@abc.com)`**
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:59]:
```suggestion
    * Configure `APPSMITH_SIGNUP_ALLOWED_DOMAINS` with a second-level domain name (eg: `abc.com`), to only allow users with an email address from that domain name to log in (eg: `john@abc.com`). You can provide several domain names using a comma-separated list.
```

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 13]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
-      * **Note**: you can provide several domain names using a comma-separated list
+    * **Note**: you can provide several domain names using a comma-separated list
     * [Restart the Appsmith instance](../)
 
 ---
 
 :::caution
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:65]: this callout should be on the top right after the heading 'Configure Appsmith instance'

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md [hunk 14]
```diff
@@ -7,64 +7,63 @@ sidebar_position: 1
...
+The Client ID and Client Secret entered aren't verified by any means. Please test it out manually before disabling the Form login authentication method. If Form login is disabled before testing the Google authentication configuration setup, it may lead to the user getting stuck on the login screen if the Google configuration is invalid. You’ll have to update the environment variables manually to fix this, as you can't edit it via the Admin settings page since you won’t be logged in anymore.
 :::
 
-After these steps, Google Login should now be enabled for your Appsmith installation. If you are running appsmith behind an ELB / Proxy, please configure Nginx [following this guide](/help-and-support/troubleshooting-guide/deployment-errors#oauth-sign-up-not-working).
+After these steps, Google Login should be enabled for your Appsmith installation. If you are running appsmith behind an ELB / Proxy, please configure Nginx [following this guide](/help-and-support/troubleshooting-guide/deployment-errors#oauth-sign-up-not-working).
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:69]: the nginx bit should be bullet 2 in caution caution

**Review by danciaclara — CHANGES_REQUESTED** [review]
can we also crop the images so the white space is not unecessarily wasting our space on the page?

Screenshot 2023-03-09 at 19 36 25
Screenshot 2023-03-09 at 19 36 17

File: website/docs/getting-started/setup/instance-configuration/authentication/google-login.md (hunk 2 above)
Comment (Pranay105) [reply website/docs/getting-started/setup/instance-configuration/authentication/google-login.md:15]: I think that's not the goal here. We want to tell the users how to configure OAuth. If a user comes to this page, he'll already know that he need to have a project.

**Review by danciaclara — APPROVED** [review]

**Pull request merged**